src/crystal/system/unix/file_descriptor.cr

```diff
@@ -252,6 +252,19 @@ module Crystal::System::FileDescriptor
     io
   end
 
+  def self.write_fully(fd : LibC::Int, ptr : Pointer, size : Int32)
+    write_fully Slice.new(ptr.as(UInt8*), size)
+  end
```

Comment:
**polish:** I think this signature is a bit misleading about the pointer and size parameters. The the size refers to the bytesize of the pointer value, not the number of values in that pointer (as in `Slice.new(pointer : Pointer(T), size : Int)`).

And we actually don't need `size` here because it can be derived from `sizeof(ptr.value)`.
So maybe we could simplify the def like that
```suggestion
  def self.write_fully(fd : LibC::Int, ptr : Pointer, size : Int32 = 1)
    write_fully Slice.new(ptr, size).unsafe_slice_of(UInt8)
  end
```

Further down we can skip the size argument except for `message.bytesize`, although that usage could pass `message.to_slice` directly, so maybe the `size` parameter here is unnecessary.